### PR TITLE
Removed SharedServer

### LINKED ISSUE #14989: Migrate servlet-3.1 tests away from Shared Server
Update tests for *com.ibm.ws.webcontainer.servlet.3.1_fat* to stop using the deprecated Shared Server.

JSPServerHttpUnit was done here: https://github.com/OpenLiberty/open-liberty/issues/16449
VHServerHttpUnit was  done here: https://github.com/OpenLiberty/open-liberty/issues/16452
HttpSessionAttListener was done here: https://github.com/OpenLiberty/open-liberty/issues/16468
WCServerHttpUnit was done here: https://github.com/OpenLiberty/open-liberty/issues/16478
UpgradeWrite/ReadListenerHttpUnit was done here: https://github.com/OpenLiberty/open-liberty/issues/16557
UpgradeReadWriteTimeoutHttpUnit  was done here: https://github.com/OpenLiberty/open-liberty/issues/16568
AsyncRead/WriteListenerHttpUnit was done here: https://github.com/OpenLiberty/open-liberty/issues/16638
FormLoginReadListenerTest was done here: https://github.com/OpenLiberty/open-liberty/issues/16741
CDINoInjectionTests was done here: https://github.com/OpenLiberty/open-liberty/issues/16797
CDITests was done here: https://github.com/OpenLiberty/open-liberty/issues/17421
NBMultiReadTest was done here: https://github.com/OpenLiberty/open-liberty/issues/19935

### PR DESCRIPTION
**Pull request by tssmith04**
Fixes #14989

### CODE REVIEW

**Comment by tssmith04** [comment]
#build

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java:1]: You can remove the contributors section of the copyright header as well.

Comment (pnicolucci) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java:1]: I won't comment on each file but there are others that could be cleaned up. Thanks!

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java [hunk 2]
```diff
@@ -188,7 +181,7 @@ public String getParameterText(String... parms) {
      * @return The request url.
      */
     public String getCurrencyURL(String... parms) {
-        return CURRENCY_INTERCEPTOR_URL + getParameterText(parms);
+        return CURRENCY_INTERCEPTOR_URL_FRAGMENT + getParameterText(parms);
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java:181]: Why this change?

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIListenersTest.java [hunk 3]
```diff
@@ -129,7 +122,7 @@ public String getListenersURL(String operationName, String comment) throws Unsup
 
         // @formatter:off
         return
-            LISTENERS_SERVLET_URL + "?" +
+            LISTENERS_SERVLET_URL_FRAGMENT + "?" +
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIListenersTest.java:122]: Same question as previously, this looks like a consistent change so just want to understand why the URL changed.

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java [hunk 4]
```diff
@@ -407,19 +356,43 @@ public void test_ServletMapping() throws Exception {
         logmsg = wlp.waitForStringInLogUsingMark("SRVE9016E:");
         Assert.assertNotNull("TestServletMapping application deployment did not result in  message SRVE9016E: ", logmsg);
 
+        // Need to sleep because it times out waiting for the config update in logs while server is still doing cleanup/spinup stuff
+        Thread.sleep(5000);
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java]: Would we be able to avoid this sleep if we `waitForConfigUpdateInLogUsingMark` was used for the first config update?

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java [hunk 5]
```diff
@@ -407,19 +356,43 @@ public void test_ServletMapping() throws Exception {
...
-        logmsg = wlp.waitForStringInLogUsingMark("CWWKZ0002E:.*TestServletMappingAnno");
+        String logmsg = wlp.waitForStringInLogUsingMark("CWWKZ0002E:.*TestServletMappingAnno");
         Assert.assertNotNull("TestServletMappingAnno application should have failed to start ", logmsg);
 
         // application failed to start, verify that it is because of a duplicate servlet-mapping
         logmsg = wlp.waitForStringInLogUsingMark("SRVE9016E:");
         Assert.assertNotNull("TestServletMappingAnno application deployment did not result in  message SRVE9016E ", logmsg);
 
+        // Need to sleep because it times out waiting for the config update in logs while server is still doing cleanup/spinup stuff
+        Thread.sleep(5000);
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java]: I'd like to avoid a sleep if at all possible.

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java (hunk 2 above)
Comment (tssmith04) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIBeanInterceptorServletTest.java:181]: For the URL changes the URL variable was a combination of `BLANK_CONTEXT_ROOT + BLANK_URL_FRAGMENT` and because the private `verifyStringsInResponse` function already takes in the `contextRoot` and remaining `path` we didn't need the combined URL variable anymore.

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIListenersTest.java (hunk 3 above)
Comment (tssmith04) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIListenersTest.java:122]: Answered above.

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java (hunk 4 above)
Comment (tssmith04) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java]: Yep, this worked.

**Comment by tssmith04** [comment]
#build

**Comment by tssmith04** [comment]
#build

**Comment by tssmith04** [comment]
#libby

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java [hunk 6]
```diff
@@ -136,27 +127,24 @@ protected WebBrowser createWebBrowser() {
     protected static final boolean IS_NOT_HTTP_URL = false;
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java]: Looks like you don't need these constants.

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java [hunk 7]
```diff
@@ -248,30 +231,24 @@ public void implTestCDINoUpgrade() throws Exception {
...
 
         logStart(methodName, testName);
-
-        WebBrowser browserS1R1 = createWebBrowser();
-        browserS1R1.setAutoRedirect(false);
-        browserS1R1.setAcceptCookies(true);
-
+        
+        HttpClientParams params = new HttpClientParams();
+        params.setCookiePolicy(CookiePolicy.DEFAULT);
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java:233]: What is the default cookie policy? Does it need to be set?

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java [hunk 8]
```diff
@@ -458,22 +431,42 @@ private void closeSocket(Socket socket, String urlText) throws Exception {
 
     private static final String NULL_SESSION_ID = null;
 
-    private void displayLog(String sessionId, String[] expectedLog, String[] unexpectedLog) throws Exception {
-        WebBrowser logBrowser = createWebBrowser();
-        logBrowser.setAutoRedirect(false);
-        logBrowser.setAcceptCookies(true);
+    private void displayLog(String[] expectedLog) throws Exception {
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java]: Minor comment -- I don't like the display logs method name. Makes it more confusing since it verifies the output for the test. 


Maybe call it verifyOutputMatchesExpectedLog or something similar?

**Review by volosied — DISMISSED** [review]
Looks good to me otherwise. Approved!

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java (hunk 7 above)
Comment (tssmith04) [reply dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CDIUpgradeHandlerTest.java:233]: It just makes sure cookies are enabled. The default behavior depends on the browser being used I believe, but it makes sure that cookies can be sent.

**Comment by tssmith04** [comment]
#build

**Review by pnicolucci — APPROVED** [review]

**Review by volosied — APPROVED** [review]
